src/figuresClasses.ts

```diff
@@ -1,19 +1,95 @@
 export interface Figure {
-
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  // eslint-disable-next-line no-useless-constructor
+  constructor(
+    public color: Figure['color'],
+    public a: number,
+    public b: number,
+    public c: number,
+    public shape: Figure['shape'] = 'triangle',
+  ) {
+    if (
+      a.toString().length <= 0
+      || b.toString().length <= 0
+      || c.toString().length <= 0
+    ) {
+      throw new Error('all sides must be > 0');
+    }
+
+    const arrayOfSides: number[] = [];
+
+    arrayOfSides.push(a, b, c);
+    arrayOfSides.sort((first, second) => second - first);
+
+    const sumOfTwoSides: number = arrayOfSides[1] + arrayOfSides[2];
+
+    if (arrayOfSides[0] >= sumOfTwoSides) {
+      throw new Error('longest side must be >= sum of two others');
+    }
+  }
+
+  getArea(): number {
+    const halfMeter: number = (this.a + this.b + this.c) / 2;
 
+    return (
+      Math.trunc((
+        Math.sqrt(halfMeter
```

Comment:
It's better to move it to a separate variable before to make your code more descriptive